src/EduManagementLab.Api/Controllers/CoursesController.cs

```diff
@@ -153,12 +154,14 @@ public ActionResult<CourseDto> AddCourseMembership(Guid courseId, Guid userId, D
 
         [HttpPut]
         [Route("{courseId}/Memberships/{userId}")]
+        [ProducesResponseType(typeof(MembershipDto), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
         public ActionResult<Course> UpdateCourseMembershipEnrolledDate(Guid courseId, Guid userId, DateTime MembershipEnrolledDate)
         {
             try
             {
                 var courseMembership = _courseService.UpdateMembershipEnrolledDate(courseId, userId, MembershipEnrolledDate);
```

Comment:
_courseService.UpdateMembershipEnrolledDate return a Course (not a membership) so the mapping on the next row does not work.